overlord/state/notices.go

```diff
@@ -211,6 +219,29 @@ type AddNoticeOptions struct {
 
 	// Time, if set, overrides time.Now() as the notice occurrence time.
 	Time time.Time
+
+	// RepeatCheck, if set, overrides "RepeatAfter" value dynamically based
+	// on the current state.
+	//
+	// NOTE: Set/Get operation on "data" are persisted and can be used
+	// for statful tracking throughout notice's lifetime.
+	RepeatCheck func(data InternalNoticeData) (newRepeatAfter time.Duration, err error)
```

Comment:
In plan we had data here too

Reply:
I asked a question in the original discussion https://github.com/snapcore/snapd/pull/13528#discussion_r1484617049 because it seems like I misunderstood your suggestion that it would be okay to not initialize.